### PR TITLE
Random fixes corrections around

### PR DESCRIPTION
**Pull request by patrickelectric**
- Move util function to static and fix shadow variable
- Add const reference for some parameters
- Remove unnecessary constructor 
- Use QRandomGenerator over qrand

### CODE REVIEW

File: src/filemanager/filemanager.h [hunk 1]
```diff
@@ -111,7 +111,7 @@ class FileManager : public QObject
      *
      */
     struct FolderStruct {
-        FolderStruct(QDir _dir, QString _extension = QString())
+        FolderStruct(QDir _dir, const QString& _extension = QString())
```
Comment (tcanabrava) [inline src/filemanager/filemanager.h]: I'd remove that completely. FolderStruct {dir, extension}; already creates what you want.

Comment (patrickelectric) [reply src/filemanager/filemanager.h]: LCMTH, great catch

File: src/link/ping1dsimulationlink.cpp [hunk 2]
```diff
@@ -35,12 +36,12 @@ void Ping1DSimulationLink::randomUpdate()
     for (int i = 0; i < numPoints; i++) {
         float point;
         if (i < stop1) {
-            point = 0.1 * (qrand() % 256);
+            point = 0.1 * QRandomGenerator::global()->bounded(256);
```
Comment (Williangalvani) [inline src/link/ping1dsimulationlink.cpp]: why 256? couldn't you do `QRandomGenerator::global()->bounded(25.6)`?

File: src/link/ping1dsimulationlink.cpp [hunk 3]
```diff
@@ -35,12 +36,12 @@ void Ping1DSimulationLink::randomUpdate()
...
         } else if (i < stop2) {
             point
                 = 255 * ((-4.0 / qPow((stop2 - stop1), 2.0)) * qPow((i - stop1 - ((stop2 - stop1) / 2.0)), 2.0) + 1.0);
         } else {
-            point = 0.45 * (qrand() % 256);
+            point = 0.45 * QRandomGenerator::global()->bounded(256);
```
Comment (Williangalvani) [inline src/link/ping1dsimulationlink.cpp]: same here

File: src/link/ping1dsimulationlink.cpp (hunk 2 above)
Comment (patrickelectric) [reply src/link/ping1dsimulationlink.cpp]: From the doc: 
`Generates one random double in the range between 0 (inclusive) and highest (exclusive).`
Probably you'll catch it better as an [0, 256) interval that is the same as [0, 255].

Comment (patrickelectric) [reply src/link/ping1dsimulationlink.cpp]: @Williangalvani let me know if this resolve the conversation

Comment (Williangalvani) [reply src/link/ping1dsimulationlink.cpp]: The code says the function is equivalent to `generateDouble() * highest`, so I thought you just move that 0.1 inside. I understand that double and integer generation will yeld different results, but I thought it didn't matter as much here.

I'm not really concerned about this, though, marking as resolved.

**Comment by jaxxzer** [comment]
lgtm

**Pull request merged**